applications/dashboard/modules/class.sitetotalsmodule.php

```diff
@@ -2,58 +2,144 @@
+
+    /**
+     * SiteTotalsModule constructor.
+     */
     public function __construct() {
         parent::__construct();
         $this->_ApplicationFolder = 'dashboard';
     }
 
+    /**
+     * @return string
+     */
     public function assetTarget() {
         return 'Panel';
     }
 
-    protected function _GetData() {
-        $px = Gdn::database()->DatabasePrefix;
-        $sql = "show table status where Name in ('{$px}User', '{$px}Discussion', '{$px}Comment')";
+    /**
+     * Get total counts
+     *
+     * @return mixed
+     */
+    public function getTotals() {
+        // check recalculate flag
+        $recalculateFlag = Gdn::cache()->get(self::RECALCULATE_KEY);
 
-        $result = ['User' => 0, 'Discussion' => 0, 'Comment' => 0];
-        foreach ($result as $name => $value) {
-            $result[$name] = $this->getCount($name);
+        if ($recalculateFlag === Gdn_Cache::CACHEOP_FAILURE) {  // expired
+            $this->tryRecalculate();
         }
-        $this->setData('Totals', $result);
+
+        // get totals from cache
+        $totals = Gdn::cache()->get(self::COUNTS_KEY);
+
+        return $totals;
     }
 
-    protected function getCount($table) {
-        // Try and get the count from the cache.
-        $key = "$table.CountRows";
-        $count = Gdn::cache()->get($key);
-        if ($count !== Gdn_Cache::CACHEOP_FAILURE) {
-            return $count;
+    /**
+     * Attempt to implement lock system. A failure likely means the
+     * cache key already exists, which would mean the lock is already in place.
+     */
+    private function tryRecalculate() {
+        $lock = Gdn::cache()->get(self::LOCK_KEY);
+
+        if ($lock === Gdn_Cache::CACHEOP_SUCCESS) { //already locked
+            return false;
+        } else {
+            $added = Gdn::cache()->add(self::LOCK_KEY, mt_rand(0, 999999), [Gdn_Cache::FEATURE_EXPIRY => self::LOCK_INTERVAL]);
+
+            if ($added) {
+                /** @var Vanilla\Scheduler\SchedulerInterface $scheduler */
+                $scheduler = Gdn::getContainer()->get(Vanilla\Scheduler\SchedulerInterface::class);
+                $scheduler->addJob(
+                    Vanilla\Scheduler\Job\CallbackJob::class,
+                    [
+                        "callback" => function () {
+                            $this->getAllCounts();
+                        }
+                    ]
+                );
+            }
+        }
+    }
+
+    /**
+     * Fire counts request to the DB
+     */
+    private function getAllCounts() {
+        $counts = ['User' => 0, 'Discussion' => 0, 'Comment' => 0];
+
+        foreach ($counts as $name => $value) {
+            $counts[$name] = $this->getCount($name);
         }
 
-        // The count wasn't in the cache so grab it from the table.
+        // cache counts
+        Gdn::cache()->store(self::COUNTS_KEY, $counts, [Gdn_Cache::FEATURE_EXPIRY => self::CACHE_TTL]);
```

Comment:
Before storing this, you probably want to iterate over `$counts` and check each value for being `=== null` or `=== false` and then don't overwrite the cache if any are not valid. Otherwise you'll break the cached data.

So you'd bail out of this method and not overwrite the COUNTS_KEY and not re-set the RECALCULATE_KEY.